test/E2E/EndToEndTest.cs

```diff
@@ -9,110 +9,120 @@
 using Xunit;
 using Microsoft.DotNet.Cli.Utils;
 using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Microsoft.DotNet.Tools.Publish.Tests;
 
-namespace ConsoleApplication
+namespace Microsoft.DotNet.Tests.EndToEnd
 {
-    public class E2ETest
+    public class EndToEndTest : TestBase
     {
         private static readonly string EXPECTED_OUTPUT = "Hello World!" + Environment.NewLine;
-        private static readonly string TESTDIR_NAME = "hellotest";
+        private static readonly string TESTDIR_NAME = "e2etestroot";
         private static readonly string OUTPUTDIR_NAME = "testbin";
-
-        private static string RootPath { get; set; }
+        
+        private string Rid { get; set; }
         private string TestDirectory { get; set; }
+        private string TestProject { get; set; }
         private string OutputDirectory { get; set; }
-        private string Rid { get; set; }
 
         public static void Main()
         {
             Console.WriteLine("Dummy Entrypoint.");
         }
        
-        public E2ETest()
+        public EndToEndTest()
         {
-            if (RootPath == null)
-            {
-                RootPath = Directory.GetCurrentDirectory();
-            }
-
-            TestDirectory = Path.Combine(RootPath, TESTDIR_NAME);
-            OutputDirectory = Path.Combine(RootPath, OUTPUTDIR_NAME);
+            TestSetup();
 
             Rid = RuntimeIdentifier.Current;
         }
 
         [Fact]
-        public void TestDotnetCompile()
+        public void TestDotnetBuild()
         {
-            TestSetup();
+            var buildCommand = new BuildCommand(TestProject, output: OutputDirectory);
+
+            buildCommand.Execute().Should().Pass();
 
-            TestRunCommand("dotnet", $"build -o {OutputDirectory}");
-            TestOutputExecutable(OutputDirectory);
+            TestOutputExecutable(OutputDirectory, buildCommand.GetOutputExecutableName());
         }
 
         [Fact]
-        public void TestDotnetCompileNativeRyuJit()
+        public void TestDotnetBuildNativeRyuJit()
         {
             if(SkipForOS(OSPlatform.Linux, "https://github.com/dotnet/cli/issues/527"))
             {
```

Comment:
Lets us something like [this](https://github.com/dotnet/corefx/blob/master/src/System.Globalization.Extensions/tests/IdnMapping/UseStd3AsciiRules.cs#L53) or [this](https://github.com/dotnet/cli/blob/master/test/Microsoft.DotNet.Tools.Publish.Tests/Microsoft.DotNet.Tools.Publish.Tests.cs#L73).